sktime/forecasting/compose/_category_compositor.py

```diff
@@ -0,0 +1,479 @@
+            else:
+                self.chosen_forecaster_ = copy.deepcopy(self.fallback_forecaster)
+
+        else:
+            self.chosen_forecaster_ = copy.deepcopy(self.forecasters[self.category_])
+
+        # fitting the forecaster!
+        self.chosen_forecaster_.fit(y=y, X=X, fh=fh)
+
+        return self
+
+    def _predict(self, fh, X):
+        """Forecast time series at future horizon.
+
+        private _predict containing the core logic, called from predict
+
+        State required:
+            Requires state to be "fitted".
+
+        Accesses in self:
+            Fitted model attributes ending in "_"
+            self.cutoff
+
+        Parameters
+        ----------
+        fh : guaranteed to be ForecastingHorizon or None, optional (default=None)
+            The forecasting horizon with the steps ahead to to predict.
+            If not passed in _fit, guaranteed to be passed here
+
+        X : sktime time series object, optional (default=None)
+            guaranteed to be of an mtype in self.get_tag("X_inner_mtype")
+            Exogeneous time series for the forecast
+
+        Returns
+        -------
+        y_pred : sktime time series object
+            should be of the same type as seen in _fit, as in "y_inner_mtype" tag
+            Point predictions
+        """
+        # Obtain the prediction values for the given horizon.
+        y_pred = self.chosen_forecaster_.predict(fh=fh, X=X)
+
+        return y_pred
+
+    def _update(self, y, X=None, update_params=True):
+        """Update time series to incremental training data.
+
+        private _update containing the core logic, called from update
+
+        State required:
+            Requires state to be "fitted".
+
+        Accesses in self:
+            Fitted model attributes ending in "_"
+            self.cutoff
+
+        Writes to self:
+            Sets fitted model attributes ending in "_", if update_params=True.
+            Does not write to self if update_params=False.
+
+        Parameters
+        ----------
+        y : sktime time series object
+            guaranteed to be of an mtype in self.get_tag("y_inner_mtype")
+            Time series with which to update the forecaster.
+            if self.get_tag("scitype:y")=="univariate":
+                guaranteed to have a single column/variable
+            if self.get_tag("scitype:y")=="multivariate":
+                guaranteed to have 2 or more columns
+            if self.get_tag("scitype:y")=="both": no restrictions apply
+        X :  sktime time series object, optional (default=None)
+            guaranteed to be of an mtype in self.get_tag("X_inner_mtype")
+            Exogeneous time series for the forecast
+        update_params : bool, optional (default=True)
+            whether model parameters should be updated
+
+        Returns
+        -------
+        self : reference to self
+        """
+        self.chosen_forecaster_.update(y=y, X=X, update_params=update_params)
```

Comment:
this should also be explained in the docstring, i.e., that the transformation is not updated.

or, we could make it a parameter whether the transformation updates, but perhaps not in this PR (seems fiddly)

Reply:
I agree that it might be good to add that as a parameter (if the transformer updates, and be extension the chosen forecaster updates too). I'll explain this in the docstring for now but let me know if you'd like me to add this in for this PR!